### DISCUSSION TITLE
User authorization using netbox-plugin-auth-saml2 SSO plugin

### DISCUSSION THREAD

**Discussion by dan-tabarca** [netbox-community/netbox#8468] [General]
I am using netbox-plugin-auth-saml2 to authenticate with SSO against active directory. Authentication works well, but I would need to do authorizations per type of user. 
Where should be defined the groups? Somewhere in Netbox configuration file, in an external SAML2 URL or in the active directory? 

Thank you.

**Comment by markkuleinio** [comment]
See https://github.com/jeremyschulman/netbox-plugin-auth-saml2, there is an example that maps a group "saml-group3" (from SAML response) to NetBox group "netbox-group". Then you use the "netbox-group" to assign the desirable permissions to that group in NetBox permissions configuration (in the NetBox admin section, not in `configuration.py`).

This all assumes that the "groups" claim (as specified in GROUP_ATTR) is included in the SAML response.

Note that I've only used it with Azure AD, so if you are talking about on-premises AD, something can be different.
